src/java.base/share/classes/sun/nio/ch/ChannelInputStream.java

```diff
@@ -53,13 +53,26 @@ class ChannelInputStream extends InputStream {
     private byte[] bs;       // Invoker's previous array
     private byte[] b1;
 
+    private Boolean isOther = null;
```

Comment:
I don't particularly like using a Boolean for tri-states but it's not too bad here. No need to initialize to null. It could be Stable but probably not much benefit here all usages require file I/O that dominates.

Are you going to add a comment to this field as readers might know now what "other" means? In the APIs we say "something other than a regular file, directory, or symbolic link" and maybe that could be useful here.

Reply:
Fixed in a163d8d and 492659f.